### ISSUE TITLE
Add configurable upload timeout

### ISSUE THREAD

**Issue by robjloranger** [gomods/athens#207]
Just an environment variable like `ATHENS_STORAGE_TIMEOUT` which should be set in seconds.

Each driver implementing the `(./pkg/storage).Saver` interface should be updated to use this value. Should be 300 seconds by default when no variable is set.

This will be cleaner to implement once #190 is complete.

**Comment by robjloranger** [comment]
Sweet, yep. Now just the other drivers need to be updated to use it

**Comment by michalpristas** [comment]
yes, i will have a separate PR for that and more soon.
i did not want to mix concerns in s3 pr
